src/main/java/com/epam/poc/commons/BasePage.java

```diff
@@ -247,5 +250,26 @@ public void switchToFirstTab(WebDriver driver) {
 
     public void closeCurrentTab(WebDriver driver) {
         driver.close();
+    }    
+
+    public String getNewTabUrl(WebDriver driver) {
+        Set<String> handles = driver.getWindowHandles();
+        for (String actual : handles) {
+            if(!actual.equalsIgnoreCase(this.getCurrentUrl(driver))){
+                driver.switchTo().window(actual);
+                waitForPageLoadedCompletely(driver);
+            }
+        }
+        return driver.getCurrentUrl();
+    }
+
+    public <Page extends BasePage> Page hover(WebDriver driver, WebElement webElement){
```

Comment:
This method already existed in BasePage class with the name "hoverElement()".
Please try to use the existed methods

Reply:
yes, thank you, now I see in the recently merged commit, lemme reuse it